docs/content/developers/design-keuzes.md

```diff
@@ -121,14 +121,12 @@ De URLs in dit voorbeeld zijn uiteraard fictief.
   ```
 
 
-## Gerelateerde objecten zonder eigen resource / Groepsattributen
+## Groepsattributen
 
-Indien een (hoofd)object een gerelateerd object (of lijst van objecten) heeft,
-dat **geen** eigen resource URL nodig heeft, dan wordt het (child)object inline
-(binnen het hoofdobject) ontsloten. Deze child objecten zijn binnen RGBZ
-ook wel gedefinieerd als *groepsattributen*. Typische voorbeelden zijn
-(zaak-)kenmerken of (zaak-)eigenschappen waarin `Zaak` het hoofdobject betreft,
-en `Kenmerk` en `Eigenschap` de child objecten.
+Indien een object een groepattribuutsoort heeft (een groep van bij elkaar behorende attribuutsoorten), al dan niet herhalend d.w.z. kardinaliteit 1 of meer (een 'lijst' van waarden van de attributen van het groepattribuut), dan wordt het groepattribuut inline
```

Comment:
M.i. kan dit beter to-the-point geformuleerd worden als:

> Indien een object een groepsattribuutsoort heeft (of een lijst van deze), dan wordt het groepsattribuut inline (binnen het object) ontsloten.

Reply:
Ik wilde eigenlijk van het concept "groep(s)attribuut(soort)" af en noemde het voornamelijk om de relatie met het RGBZ te verduidelijken maar wilde dat niet de primaire definitie maken. Op database niveau kan dit namelijk vaak als apart tabelletje worden geimplementeerd, waarvan ik dan duidelijk wilde maken dat die niet via CRUD ontsloten hoeft te worden.

> Wellicht wil je als developer alleen de ontwerpkeuze weten. Ik wil ook graag het waarom daarvan weten. Zodat later te achterhalen is wat de reden van de keuze is.

Quote van Arjan in deze PR, maar ook ik wil weten waarom iets inline wordt gedaan. Als de enige reden is "omdat het een groepsattribuut betreft", vind ik dat niet voldoende, dan moeten we dat ook even uitleggen. Als we dat doen, vind ik het prima.

> Dit is een rare en m.i. onjuiste ontwerpkeuze. Er is, in het RGBZ, geen sprake van "een (hoofd)object [dat] een gerelateerd object (of lijst van objecten) heeft, dat geen eigen resource URL nodig heeft". Dit is een misvatting. Het RGBZ kent objecttypen die relaties met elkaar hebben. Indien een objecttype een eigen bestaansrecht heeft dan vraagt dat om uitwerking in een eigen resource. Kenmerken van dat objecttype maken deel uit van die resource. Kenmerken zijn attribuutsoorten. En die kunnen genest zijn: groepattributen. Maar ook dat blijven kenmerken van het objecttype, Dus geen enkele reden om dat uit te werken in een eigen resource.

Ik begrijp niet zo goed wat je zegt. Je geeft aan dat de design keuze nu een misvatting is, maar volgens mij ben ik het verder eens met je wat zegt en staat dat er ook. Kenmerken worden ook niet uitgewerkt als eigen resource?


Er zijn echter HEEL VEEL groepsattribuutsoorten en ze moeten wellicht niet allemaal op dezelfde manier behandeld worden. Dit komt denk ik deels doordat er niet altijd (duidelijk) sprake is van een relatie in het RGBZ, als we dat aanhouden tegen het concept van linked data. Dit wordt dan opgelost met groepsattribuutsoort.

2 voorbeelden:

![image](https://user-images.githubusercontent.com/96970/44718900-e3ad5e00-aac1-11e8-836e-1322d6ea9e55.png)
_- `Kenmerken` binnen Objecttype Zaak uit RGBZ 2.0 pagina 76/77_

Kenmerken worden in de API dus WEL inline in het resource ontsloten. Het heeft geen enkel nu om  `/api/v1/kenmerken/` aan te bieden, en er wordt ook niet door andere resource verwezen naar kenmerken. Volgens mij helemaal conform Arjan's argumentatie.

![image](https://user-images.githubusercontent.com/96970/44718959-21aa8200-aac2-11e8-90e3-cc2023fab6d1.png)
_- `Gerelateerde externe Zaak` binnen Objecttype Zaak uit RGBZ 2.0 pagina 78_

Dit lijkt me nu typisch een voorbeeld van een relatie naar een (externe) Zaak, en dus naar een eigen resource en niet inline. Terwijl het dus een groepsattribuutsoort betreft. Hoewel de zaak extern is zie ik geen `bronorganisatie` veld, waarmee de zaak ook niet uniek geidentificeerd kan worden.

Mocht dit laatste voorbeeld niet validate zijn, dan zie ik `Rol.Afwijkend binnenlands correspondentieadres` of andere adres groepsattribuutsoorten als voorbeeld. Sommige groepsattributen vind ik ook onzinnig: `Bestandsnaam` (optioneel) bijvoorbeeld die bestaat uit `bestandsnaam` (verplicht) en `extensie` (optioneel). Denk dat het juist wel aardig is om niet klakkeloos het RGBZ in deze over te nemen maar dat er per groepsattribuutsoort even over nagedacht wordt.

@ArjanKloosterboer Of vind jij wel dat bovenstaande voorbeelden allemaal inline zouden moeten zijn?

Reply:
Volgens mij zijn we ergens anders al tot de conclusie gekomen dat 'Gerelateerde externe zaak' een alternatief is voor de relatie naar die externe zaak in het geval dat die externe zaak niet via een API bereikbaar is. Dus ook dit is een groepattribuut die inline ontsloten wordt. 
Conclusie: elke groepattributsoort wordt inline ontsloten.